components/electric/lib/electric/satellite/client_reconnection_info.ex

```
@@ -446,23 +568,67 @@ defmodule Electric.Satellite.ClientReconnectionInfo do
           # Client has seen this additional data blob!
           diff = :ets.lookup_element(@additional_data_ets, key, 2)
           graph = merge_in_graph_diff(graph, diff)
-          :ets.delete(@additional_data_ets, key)
-          clear_stored_actions(client_id, covered_txns)
+          :ok = delete_additional_data(origin, key)
+          clear_stored_actions(origin, client_id, covered_txns)
           # See if we can do one more
-          advance_by_additional_data(graph, client_id, acknowledged)
+          advance_by_additional_data(graph, client_id, acknowledged, origin)
         else
           # Client hasn't seen this, so we're done here
           graph
         end
     end
   end
 
+  @delete_additional_data_for_key_query """
+  DELETE FROM
+    #{Extension.client_additional_data_table()}
+  WHERE
+    client_id = $1 AND ord = $2
+  """
+
+  defp delete_additional_data(origin, {client_id, _xmin, order, _subject, _subscription_id} = key) do
+    :ets.delete(@additional_data_ets, key)
+
+    Client.pooled_query!(origin, @delete_additional_data_for_key_query, [client_id, order])
+
+    :ok
+  end
+
+  @insert_subscription_query """
+  INSERT INTO
+    #{Extension.client_shape_subscriptions_table()}(
+      client_id,
+      subscription_id,
+      min_txid,
+      ord,
+      shape_requests
+    )
+  VALUES
+    ($1, $2, $3, $4, $5)
+  ON CONFLICT
+    (client_id, subscription_id)
```

Why an `ON CONFLICT` clause at all here? This should not be reachable at all under current model, so I'd rather it fail visibly when we do horizontal scaling. I'm not sure just taking a later write here is correct in a general case. This applies to other queries as well

---

Good catch. This is a leftover from my initial attempt to store additional data for subscriptions in the `client_subscriptions` table. Removed.